src/driver/main.cpp

```diff
@@ -180,11 +255,25 @@ struct loader
             if(file_type == "onnx")
             {
                 onnx_options options;
-                options.default_dim_value      = batch;
+                if(default_dyn_dim.empty())
+                {
+                    options.default_dim_value = batch;
+                }
+                else
+                {
+                    options.default_dyn_dim_value = parse_dyn_dim_str(default_dyn_dim);
+                }
                 options.skip_unknown_operators = skip_unknown_operators;
                 options.print_program_on_error = true;
-                options.map_input_dims         = map_input_dims;
-                p                              = parse_onnx(file, options);
+                if(map_dyn_input_dims.empty())
+                {
+                    options.map_input_dims = map_input_dims;
```

Comment:
users can only use one of the other of `map_input_dims` & `map_dyn_input_dims`.   Need to error if they supply both.